### PR TITLE
Refactor ip_detection_extensions to avoid cross extension dependencies.

### PR DESCRIPTION
**Pull request by KBaichoo**
Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

Commit Message:Refactor ip_detection_extensions to avoid cross extension dependencies.
Additional Description:
Risk Level: n/a test only
Testing: ran unit test
Docs Changes: NA
Release Notes: NA
Platform Specific Features: NA

I thought about putting this in the corresponding test/extension directories, but that'd require changing visibility of the packages which I haven't found elsewhere in test/extensions.

### CODE REVIEW

**Review by rgs1 — APPROVED** [review]
Thanks for cleaning this up. LGTM once/if CI is happy.

**Comment by dmitri-d** [comment]
lgtm

**Comment by KBaichoo** [comment]
PTAL @envoyproxy/maintainers

**Review by antoniovicente — APPROVED** [review]
Thanks for the test refactor.

**Pull request merged**
